### PR TITLE
chore: use latest version of custom modules, move to ESM

### PR DESCRIPTION
**Pull request by HunteRoi**
## Description
Update dependencies (custom modules, mainly).
Move entire project from CommonJS modules to ES modules.

## Motivation and Context
A few changes in the organization at school required changes to be reflected in the modules. Also, the bot itself was way too behind in versions compared to all the modules.

## Usage examples
/

## How Has This Been Tested?
Manually, in the Beta guild. And each module is tested individually as well (either manually or by automated tests).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { Configuration } from './models/Configuration';
-import { version } from '../package.json';
+import { Configuration } from './models/Configuration.js';
+import packageInfo from '../package.json' with { type: "json" };
```
Comment (DrakeXorn) [inline src/config.ts:2]: Didn't know that. Good to know

File: src/services/PostgresDatabaseService.ts [hunk 2]
```diff
@@ -186,11 +184,11 @@ export default class PostgresDatabaseService implements IDatabaseService {
         return values.map(([prop], index) => prop + ' = $' + (index + offset)).join(', ');
     }
 
-    #deconstructValues(values: [string, any][]): Value[] {
+    #deconstructValues(values: [string, any][]): any[] {
```
Comment (DrakeXorn) [inline src/services/PostgresDatabaseService.ts:187]: Ewww, any

**Review by DrakeXorn — APPROVED** [review]
Fair enough

File: src/services/PostgresDatabaseService.ts (hunk 2 above)
Comment (HunteRoi) [reply src/services/PostgresDatabaseService.ts:187]: I am totaly with you on that regard! Apparently, ts-postgres was updated and they removed the `Value` type!
I had no intention to edit the bit in the beginning but I was forced to 😞

**Pull request merged**